components/sup/protocols/event.proto

```diff
@@ -14,6 +14,7 @@ message EventMetadata {
   string application = 4;
   string environment = 5;
   map<string, string> meta = 6;
+  string fqdn = 7;
```

Comment:
I'd prefer
```suggestion
  string fully_qualified_domain_name = 7;
```
and a comment describing which fully qualified domain name this is and when it's resolved/updated

Reply:
I agree that fqdn might not have a true meaning and would be good to use the full name as you suggest, but also now I am wondering if instead of FQDN we should just pass it as `hostname`. Thoughts @christophermaier @baumanj ??

Reply:
FQDN is a common reference to Fully Qualified Domain Name and is especially common to the various products we ship at Chef. It's even referred to as such in the `man hostname`:

```
   THE FQDN
       The FQDN (Fully Qualified Domain Name) of the system is the name that the resolver(3) returns for the host name, such as, ursula.example.com.  It is usually the hostname followed by the DNS domain name (the part  after the first dot).  You can check the FQDN using hostname --fqdn or the domain name using dnsdomainname.
```

I'm definitely for leaving this as is.

Reply:
Additionally, `hostname` and `fqdn` are not the same, so they can't be used interchangeably.

Reply:
I agree that `hostname` and `fqdn` aren't interchangeable. And for anyone that knows what `fqdn` means without any additional context, `fully_qulaified_domain_name` should be just as clear, but for those who don't, the latter is a much more informative name than the former. I'm well aware of what an FQDN is myself, but the goal is to make the code as easy to understand and reason about for as many people as possible, right? The tradeoff here is clarity versus concision. If it were a local variable, I'd be more inclined to `fqdn`, but I prefer more explicitness in field names.

As for whether what we want to store here is the FQDN or just the hostname, that's a question of what the needs are that we're trying to satisfy with this change, so I don't really know. Perhaps, @christophermaier does.

Reply:
@baumanj  that crate is abandoned.

Reply:
What makes you say that? It hasn't been updated in awhile, but that could just be that there's no need to make changes to a very simple crate that does the job it's designed to.

Reply:
I say it because it says it in the tittle of the repository.

![816596FD-11B9-4756-98CA-1A722DFD1FB8](https://user-images.githubusercontent.com/5712253/57339317-b73ced80-70f6-11e9-8188-2c3d2303ed07.jpeg)

Reply:
@baumanj Also this: (https://github.com/fengcen/hostname/pull/4#issuecomment-455735989)
>fengcen **(REPO-OWNER)** commented on Jan 18
Sorry guys. I have no time and interest in Rust right now. This project is going to Abandoned status.

Reply:
In my opinion, `fqdn` is acceptable for a field name.  The field name gets translated into a struct field name in code, so it's something you'll end up typing with some frequency.  I would venture to guess that any programmers working in this code will know what a `fqdn` is.  For those that don't, it feels like an abbreviation worth learning, because it comes up often.  Hypothetically, if there were a field on a protobuf message that indicated whether TCP was being used or not, it would be reasonable to call it `tcp` instead of `transmission_control_protocol`, even though the latter is more explicit.  This feels like a similar situation to me.  If more clarity is warranted, a comment could be added above the name indicating that it's a fully qualified domain name.  I try not to worship at the altar of brevity too much, but in this particular case, it feels like a reasonable tradeoff to me.